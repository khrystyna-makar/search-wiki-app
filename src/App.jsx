import {
  Routes,
  Route
} from 'react-router-dom';

import Home from './pages/Home';
import Search from './pages/Search';
import NotFound from './pages/NotFound';

const App = () => (
  <Routes>
    <Route exact path="/" element={<Home />} />
    <Route path="/search" element={<Search />} />
    <Route path="*" element={<NotFound />} />
  </Routes>
)

export default App