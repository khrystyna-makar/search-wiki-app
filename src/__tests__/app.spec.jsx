import { mount } from "enzyme";
import App from "../App";
import { beforeEach, describe, expect, it } from "vitest";
import { MemoryRouter } from "react-router-dom";

const render = (initialEntries) => mount(<MemoryRouter initialEntries={initialEntries}><App /></MemoryRouter>);

describe('App component', () => {
    let sut;

    describe('when home page is rendered', () => {
        beforeEach(() => {
            sut = render(['/'])
        });
        it('should match home page snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });

    describe('when not found page is rendered', () => {
        beforeEach(() => {
            sut = render(['/hdghgf'])
        });
        it('should match not found page snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });
})