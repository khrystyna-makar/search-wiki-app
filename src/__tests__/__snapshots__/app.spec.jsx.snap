// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > when home page is rendered > should match home page snapshot 1`] = `ReactWrapper {}`;

exports[`App component > when not found page is rendered > should match not found page snapshot 1`] = `ReactWrapper {}`;
